servers/src/mining/stratumserver.rs

```diff
@@ -502,7 +502,7 @@ impl StratumServer {
 			return Err(serde_json::to_value(e).unwrap());
 		}
 		// If the difficulty is high enough, submit it (which also validates it)
-		if share_difficulty >= self.current_difficulty {
+		if share_difficulty >= self.current_difficulty && !self.sync_state.is_syncing() {
```

Comment:
Could you please hold this PR for a while? 😄   to give us some time to address that 'impossible' happening found in #1632, and confirm that problem is well fixed.
I'm afraid if this fix is merged, then not easy to reproduce that root cause.

BTW, shall we log a warning message here? to remind the reason of this ignored 'submit' block.

Reply:
Good point, we can definitely wait a few days for this. But I think it's already pretty obvious from the logs when you mine a block while syncing is enabled, no?

Reply:
> But I think it's already pretty obvious from the logs when you mine a block while syncing is enabled

Yes, fully agree. 

I mean the fix in this PR is good and necessary, but the root cause for that strange **impossible** happening (https://github.com/mimblewimble/grin/issues/1632#issuecomment-426188635 - `update_header_head` return `None` but `header_head` has been updated!) could be a major risk and deserve a deep checking.

Reply:
I find it's quite difficult to reproduce that special scenario: a bunch of Headers coming and in the same time mined a new block. So giving up to try reproducing problem, and please go on to merge this fix :)